lib/broker_discovery.dart

```diff
@@ -36,20 +39,23 @@ class BrokerDiscoveryClient {
 
     var interfaces = await NetworkInterface.list();
     try {
+      var joinMulticast = _socket.joinMulticast;
       for (var interface in interfaces) {
         try {
-          _socket.joinMulticast(new InternetAddress("239.255.255.230"), interface: interface);
+          Function.apply(joinMulticast, [kBroadcastAddress, interface]);
```

Comment:
why the Function.apply?

Reply:
Earlier Dart versions make `interface` a named parameter, while newer Dart versions make it an optional parameter. We have to cover both, and this allows for warnings to be silenced since we have both versions show up in code.

I also reordered the newer Dart to be the first case rather than the old Dart, not really sure that matters much.

Reply:
I'd add that as a comment for documentation's sake. Otherwise it's very likely that we'll wonder why it's like that next time we step on it